### PR TITLE
[Nuclio] Support internal external invocation urls

### PR DESCRIPTION
**Pull request by liranbg**
This is a work in progress 🚬 

- added "unhealthy" to the function states when deployment needs to raise an error
- support internal / external invocation urls, this is a Nuclio >= 1.6.x feature. I left the `address` field for backwards compatibility for those working against nuclio 1.5.x

Invoking a function would use the `internal` invocation url if applicable and running within Kubernetes cluster. else will fallback to external url or the good-old address.

deploy function will return the external address, for backwards compatibility. the return address can be used on your jupyter notebook.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,7 +10,9 @@ kfp~=1.0.1
 nest-asyncio~=1.0
 # >=5.5 from nuclio-jupyter, <7.17 cause from 7.17 python 3.6 is not supported (and models-gpu-legacy image build fail)
 ipython>=5.5, <7.17
-nuclio-jupyter~=0.8.15
+#nuclio-jupyter~=0.8.15
+# TODO: remove and use nuclio-jupter
```
Comment (Hedingber) [inline requirements.txt:14]: Commenting so we won't forget to remove before merging

**Review by Hedingber — APPROVED** [review]
Very minor comments, approved

File: mlrun/api/api/endpoints/functions.py [hunk 2]
```diff
@@ -215,14 +215,25 @@ def build_status(
             nuclio_name,
             last_log_timestamp,
             text,
+            function_status_,
```
[the remark this answers is not in the document]
Comment (liranbg) [reply mlrun/api/api/endpoints/functions.py:218]: will change to `status`

File: mlrun/runtimes/function.py [hunk 3]
```diff
@@ -406,19 +423,36 @@ def deploy(
...
+                save_record = True
+
+        logger.info(
+            "successfully deployed function",
+            internal_invocation_urls=self.status.internal_invocation_urls,
+            external_invocation_urls=self.status.external_invocation_urls,
+        )
+
+        if save_record:
+            self.save(versioned=False)
```
[the remark this answers is not in the document]
Comment (liranbg) [reply mlrun/runtimes/function.py:454]: took into consideration

**Review by Hedingber — APPROVED** [review]

**Pull request merged**